src/components/TripDetail/TripDetail.js

```diff
@@ -1,7 +1,39 @@
-import React from 'react';
-import { Link } from 'react-router-dom';
+import React, { useEffect } from 'react';
+import { withRouter, Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import PropTypes from 'prop-types';
+import { userActions } from '@providers/user/user';
+import { tripActions } from '@providers/trip/trip';
+
+const mapStateToProps = state => {
+  return {
+    auth: state.auth,
+  };
+};
+
+const mapDispatchToProps = dispatch => {
+  return {
+    actions: {
+      user: bindActionCreators(userActions, dispatch),
+      trip: bindActionCreators(tripActions, dispatch),
+    },
+  };
+};
+
+const TripDetail = ({ auth, actions, match }) => {
+  useEffect(() => {
+    const currentTripId = match.params.tripId;
+    let inTrip = false;
+    auth.profile.trips.forEach(el => {
+      if (el.id === currentTripId) inTrip = true;
+    });
+    if (!inTrip) {
+      actions.user.updateProfile({ joinTripId: currentTripId });
+      actions.trip.updateTrip({ tripId: currentTripId });
+    }
+  }, []);
```

Comment:
This part need to restructure. For the intrip variable it should be a state instead and separate its logic to a separate useEffect

Reply:
Per my below comment, I think the intrip logic with be a helper for MapStateToProp